src/avx512bw/avx512bw-from-valid-utf8.inl.cpp

```diff
@@ -0,0 +1,345 @@
+// file included directly
+
+// File contains conversion procedure from VALID UTF-8 strings.
+
+/*
+    expand_bytes takes 16 + 4 bytes (of a UTF-8 string)
+    and loads all possible 4-byte substring into an AVX512 register.
+
+    For example if we have bytes abcdefgh... we create following 32-bit lanes
+
+    [abcd|bcde|cdef|defg|efgh|...]
+     ^                          ^
+     byte 0 of reg              byte 63 of reg
+*/
+__m512i expand_bytes(const char* ptr) {
+    // load bytes 0..15 (16)
+    const __m128i tmp0 = _mm_loadu_si128((const __m128i*)ptr);
+    const __m512i t0 = _mm512_broadcast_i64x2(tmp0);
+
+    // load bytes 16..19 (4)
+    const uint32_t tmp1 = *(uint32_t*)(ptr + 16);
```

Comment:
That's undefined behavior.

You should do 
```
memcpy(&tmp1, ptr+16, sizeof(tmp1)) 
```

(Yes. I am aware that it makes no difference, but people will complain.)

Reply:
That's a good point, fixed. It's better not to have any weird code in the lib.